### PR TITLE
Configurable autoinstrumentation log level - part1

### PR DESCRIPTION
**Pull request by lachmatt**
## Why
Towards #2183 
Towards #372

## What

Log level for autoinstrumentation configurable through `OTEL_LOG_LEVEL`.
Still to do:
- update configurations docs
- handle configured log level in loader
- handle configured log level in native code

## Tests

Included in PR.

## Checklist

~~- [ ] `CHANGELOG.md` is updated.~~
~~- [ ] Documentation is updated.~~
~~- [ ] New features are covered by tests.~~

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation/Logging/LogLevel.cs [hunk 1]
```diff
@@ -1,5 +1,9 @@
-// <copyright file="LogLevel.cs" company="OpenTelemetry Authors">
-// Copyright The OpenTelemetry Authors
+//------------------------------------------------------------------------------
```
Comment (lachmatt) [inline src/OpenTelemetry.AutoInstrumentation/Logging/LogLevel.cs:1]: update license as it seems to originate from [here](https://github.com/serilog/serilog/blob/dev/src/Serilog/Events/LogEventLevel.cs)

File: test/OpenTelemetry.AutoInstrumentation.Tests/OtelLoggingTests.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+using FluentAssertions;
+using OpenTelemetry.AutoInstrumentation.Logging;
+using Xunit;
+
+namespace OpenTelemetry.AutoInstrumentation.Tests;
+
+public class OtelLoggingTests
+{
+    [Fact]
+    public void WhenNoFileSizeIsConfigured_Then_DefaultIsUsed()
```
Comment (Kielek) [inline test/OpenTelemetry.AutoInstrumentation.Tests/OtelLoggingTests.cs:26]: Consider commonly used IDisposable tests for env. vars.

Comment (lachmatt) [reply test/OpenTelemetry.AutoInstrumentation.Tests/OtelLoggingTests.cs:31]: changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2277/commits/c912cfecbdb689d1c5cb6491c633c10cf3ed9fc4

File: test/OpenTelemetry.AutoInstrumentation.Tests/OtelLoggingTests.cs [hunk 3]
```diff
@@ -32,94 +37,59 @@ public void WhenNoFileSizeIsConfigured_Then_DefaultIsUsed()
...
-        }
+        Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", "none");
+
+        OtelLogging.GetConfiguredLogLevel().Should().BeNull();
+    }
+
+    public void Dispose()
+    {
+        UnsetLogLevelEnvVar();
+    }
```
Comment (pellared) [inline test/OpenTelemetry.AutoInstrumentation.Tests/OtelLoggingTests.cs:88]: nit: move it after constructor for better readability

Comment (lachmatt) [reply test/OpenTelemetry.AutoInstrumentation.Tests/OtelLoggingTests.cs]: changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2277/commits/c78e1f0b869a3cb19d1070bfb2d4d963b416bc0a

**Review by Kielek — APPROVED** [review]

**Review by RassK — APPROVED** [review]

**Pull request merged**